Design/Mantid_As_a _SciPy_Styled_Package.md

```diff
@@ -0,0 +1,259 @@
+    - matrix workspace math (also include `V3D`, `VMD`, ...)
+    - `Plus`, `ExponentialCorrection`, `Rebin`, `Fit`
+    - add aliases `Add`=`Plus`, `Subtract`=`Minus`
+* mantid.math.axes
+    - changing the axis of a matrix workspace (not the data), that is technique independent
+    - e.g. ScaleX`, `ConvertAxisByFormula`, `MedianBinWidth`
+* mantid.math.events
+    - deals with events (technique agnostic)
+    - `FilterEvents`, `RebinByPulseTime`
+* mantid.math.multidimensional
+    - technique agnostic multidimensional workspace operations
+    - does not include `ConvertToMD`
+* mantid.optimization
+    - Curve Fitting and Optimization
+    - e.g. Fit
+* mantid.instrument 
+    - grouping, masking, etc
+    - e.g. `GroupDetectors`, `MaskBTP`, `SetGoniometer`, `MoveInstrumentComponent`
+* mantid.workspace
+    - manipulate workspaces, history
+    - `RenameWorkspace`, `GroupWorkspaces`, `CompareWorkspaces`, `Comment`
+* mantid.metadata
+    - logs, titles, but not history
+* mantid.muons
+    - muon related algorithms and Fit Functions
+* mantid.neutrons
+    - things that are related to neutrons (time of flight), but not specific to a certain subfield (like diffraction)
+    - e.g. `ConvertUnits`, `ConvertToMD`, `NormaliseByCurrent`, `He3TubeEfficiency`
+* mantid.neutrons.crystal
+    - single crystal stuff. Will include `UnitCell`, `OrientedLattice`, `SymmetryOperation`
+    - e.g. `SetUB`, `FindPeaksMD`, `IndexPeaks`
+* mantid.neutrons.diffraction
+    - powder/amorphous diffraction stuff
+    - e.g. `StripVanadiumPeaks`, `AlignAndFocusPowder`
+* mantid.neutrons.inelastic
+    - algorithms related to both direct and indirect inelastic spectroscopy
+    - e.g. `GetEi`, `CorrectKiKf`, `SofQW`
+* mantid.neutrons.reactor
+    - single wavelength algorithms
+    - right now most are facility specific
+* mantid.neutrons.reflectometry
+    - e.g. `FindReflectometryLines`
+* mantid.neutrons.sans
+    - e.g. `CalculateEfficiency`
+* mantid.simulations
+    - deal with outside simulation programs (CASTEP, SASSENA, ...)
+    - `CalculateInelasticScatteringFromAbInitioPhonon` (`Abins`)
+* mantid.constants
+    - no algorithms here
+    - physical constants
+    - neutronic constants
+* mantid.remote
+    - `SubmitRemoteJob`, `AbortRemoteJob`
+
+**Facility Specific Libraries**
+
+* mantid.ess
+* mantid.hfir
+* mantid.ill
+* mantid.isis
+* mantid.sns
+* can add instrument or technique specific sublibraries
+    - `mantid.sns.corelli.CrossCorelation`
+
+### Generation ####
+
+The generation of the Mantid Python API for algorithms is currently done at run time on start-up, and has been found to take a large proportion of the time taken to start Mantid.  
+
+#### C++ Libraries of Algorithms ####
+
+Once C++ libraries are compiled within Mantid they will not change (other than being added to or removed from the plugins directory).  Therefore the API for a C++ library can be calculated once, and then reused until a new version that library is installed.
```

Comment:
I don't understand what this is trying to say. 

For non-plugins, `Kernel`, `API`, `Geometry`, `Beamline`,`Crystal`.... IMHO we should evaluate whether the current dynamic library approach is actually a good one.